src/test/java/BaseTest.java

```diff
@@ -1,11 +1,132 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.edge.EdgeOptions;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.time.Duration;
+import java.util.HashMap;
 
 public class BaseTest {
+    public static WebDriver driver = null;
+    public static String url = "https://qa.koel.app/";
+    public static WebDriverWait wait = null; //ADDED THIS HW20
+    public static Actions actions = null;
+
+
+    public void navigateToPage(){
+        driver.get(url);
+    }
+
+
+    //@BeforeSuite
+    //static void setupClass() {WebDriverManager.chromedriver().setup();} REMOVED HW24
+
+    public static void enterPassword(String password) {
+        WebElement passwordField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[type='password']")));
+        passwordField.click();
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+
+    public static void enterEmail(String email) {
+        WebElement emailField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[type='email']")));
+        emailField.click();
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+
+    public static void openLoginUrl() {
+        String url = "https://qa.koel.app/";
+        driver.get(url);
+    }
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

Move these methods to the corresponding class according to the POM.

![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/1bd61fbc-d13e-41f3-9ff1-3e5161a544a5)